android/src/main/java/com/swmansion/rnscreens/ScreenStack.java

```diff
@@ -240,6 +256,13 @@ protected void performUpdate() {
     }
     // animation logic end
 
+    if (mDismissed.isEmpty() && mTopScreen != null && isSlideFromBottom(mTopScreen) && visibleBottom == null) {
+      // When going forward in `slide_from_bottom` transition, we want the previous screen, which is mTopScreen now,
+      // to stay visible during transition, we remove it after the transition ended
```

Comment:
I'd love to see here "we remove it in XYZ method after...". That would pinpoint where exactly it is being removed and then everything would be clear.